### PR TITLE
move some auth stuff around and add types to secureadminconsole

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- adds types to SecureAdminConsole
- moves SecureAdminConsole into Auth feature folder 
- updates references to SecureAdminConsole

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/53446/in-helm-managed-mode-user-is-prompted-to-upload-license

#### Does this PR introduce a user-facing change?
none- this is just a refactor 

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/features/Auth/components/SecureAdminConsole.tsx [hunk 1]
```diff
@@ -56,7 +88,7 @@ class SecureAdminConsole extends React.Component {
   };
 
   validatePassword = () => {
-    if (!this.state.password || this.state.password.length === "0") {
+    if (!this.state.password || this.state.password.length === 0) {
```
Comment (rianfowler) [inline web/src/features/Auth/components/SecureAdminConsole.tsx:91]: `length` will always be a number AFAIK so it should check against `0` not `"0"`

File: web/src/features/Auth/components/SecureAdminConsole.tsx [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+    method: string;
+  } | null;
+};
+type LoginResponse = {
+  token: string;
+  expires?: number;
+  sessionRoles: string;
+};
+class SecureAdminConsole extends React.Component<Props, State> {
+  loginText: React.RefObject<HTMLDivElement>;
```
Comment (miaawong) [inline web/src/features/Auth/components/SecureAdminConsole.tsx:34]: can you explain this? I'm learning :)

File: web/src/features/Auth/components/SecureAdminConsole.tsx [hunk 3]
```diff
@@ -1,12 +1,39 @@
 import * as React from "react";
 import Helmet from "react-helmet";
-import { Utilities, dynamicallyResizeText } from "../utilities/utilities";
-import Loader from "./shared/Loader";
-import ErrorModal from "./modals/ErrorModal";
+import { Utilities, dynamicallyResizeText } from "@src/utilities/utilities";
+import Loader from "@src/components/shared/Loader";
+import ErrorModal from "@src/components/modals/ErrorModal";
 import "../scss/components/Login.scss";
```
Comment (miaawong) [inline web/src/features/Auth/components/SecureAdminConsole.tsx:6]: some fixes is needed for Login.scss import in SecureAdminConsole, check build-web logs

**Review by miaawong — COMMENTED** [review]
let me know when you've made changes. thanks!

File: web/src/features/Auth/components/SecureAdminConsole.tsx (hunk 2 above)
Comment (rianfowler) [reply web/src/features/Auth/components/SecureAdminConsole.tsx:34]: `loginText` gets assigned in the constructor to a ref: 

````
this.loginText = React.createRef();
```

For some reason TS wouldn't implicitly type it so I had to explicitly type it with this statement. 

this assigns a type to `this.loginText` but without any assignment. I think TS removes this statement when things get compiled so it doesn't do anything in the js.

Comment (miaawong) [reply web/src/features/Auth/components/SecureAdminConsole.tsx:34]: thanks so much! i actually just ran into this in a file I'm working on!! super helpful!

**Review by miaawong — APPROVED** [review]

**Pull request merged**